strongbox-commons/src/main/java/org/carlspring/strongbox/io/MultipleDigestInputStream.java

```diff
@@ -232,28 +232,25 @@ public void reload()
     public void reposition()
             throws IOException
     {
-        if (byteRanges != null && !byteRanges.isEmpty() && currentByteRangeIndex < byteRanges.size())
+        if (byteRanges != null && !byteRanges.isEmpty() && currentByteRangeIndex < byteRanges.size() && currentByteRangeIndex < byteRanges.size())
```

Comment:
As a reminder -- would you be so kind and sign the ICLA as described [here](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md)? Thanks! :)